rustls/src/client/client_conn.rs

```diff
@@ -663,6 +663,37 @@ impl ConnectionCore<ClientConnectionData> {
     }
 }
 
+/// Unbuffered version of `ClientConnection`
+///
+/// See the [`crate::unbuffered`] module docs for more details
+pub struct UnbufferedClientConnection {
+    inner: UnbufferedConnectionCommon<ClientConnectionData>,
+}
+
+impl Deref for UnbufferedClientConnection {
+    type Target = UnbufferedConnectionCommon<ClientConnectionData>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl DerefMut for UnbufferedClientConnection {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.inner
+    }
+}
```

Comment:
These should be below the inherent impl of `UnbufferedClientConnection` (c.f. [ordering for a given type](https://github.com/rustls/rustls/blob/main/CONTRIBUTING.md#ordering-for-a-given-type))